service/surf/apps/users/views.py

```diff
@@ -83,3 +84,10 @@ def get(self, request, *args, **kwargs):
         if "permissions" in request.session:
             del request.session["permissions"]
         return Response({'token': token.key})
+
+
+class DeleteAccountAPIView(APIView):
+    def post(self, request, *args, **kwargs):
+        user = User.objects.get(id=request.user.id)
```

Comment:
Je hoeft nu niet authenticated te zijn om deze API aan te roepen lijkt het. Dat zou betekenen dat request.user.id None is en dat levert dan een 404 op. Je kunt permission_classes toevoegen, om aan te geven welk type user toegang heeft en alleen authenticated users toegang geven: https://www.django-rest-framework.org/api-guide/permissions/

Waarschijnlijk moet je op de frontend dan 401 afvangen voor het geval een user wordt uitgelogd door het systeem en de pagina in de browser open blijft staan.